### DISCUSSION TITLE
Time-based cache revalidation does not work when calling fetch with manually created Request object

### DISCUSSION THREAD

**Discussion by Soberia** [vercel/next.js#69118] [Help] [answered]
### Link to the code that reproduces this issue

https://github.com/o92/next-admin

### To Reproduce

1. Send API request using `fetch()` in the following Server Component:

```javascript

export default async function Home() {
  const request = new Request('https://example.com', {next: {revalidate: 5}) // or `NextRequest`
  const response = await fetch(request)
}
```

2. Request should revalidated after 5 seconds but it never will.
3. Specifying the `Request`'s arguments directly to the `fetch()` will work as expected.

### Current vs. Expected behavior

Current:
The revalidation does not happen in the specified time range or any other time afterward.

Expected:
The `fetch()` should re-run after 5 seconds for the new requests.  

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #40-Ubuntu SMP PREEMPT_DYNAMIC Fri Jul  5 10:34:03 UTC 2024
  Available memory (MB): 31963
  Available CPU cores: 12
Binaries:
  Node: 20.16.0
  npm: 
  Yarn: 
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-rc.0 // Latest available version is detected (15.0.0-rc.0).
  eslint-config-next: 15.0.0-rc.0
  react: 19.0.0-rc-f994737d14-20240522
  react-dom: 19.0.0-rc-f994737d14-20240522
  typescript: 5.5.4
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Runtime

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by sajan018** [comment, marked as answer]
``const response = await fetch('https://api.example.com/data', {
  method: 'GET',
  headers: {
    'Content-Type': 'application/json',
  },
  next: { revalidate: 60 }, // Revalidate cache after 60 seconds
});

const data = await response.json();

**Reply by sajan018** [reply]
I hope it is working?

**Reply by Soberia** [reply]
Yes, it's working. I mentioned in part 3 of **To Reproduce** that it works if arguments are passed directly to the `fetch()` function. The problem is it's not possible to pass the `Request` object to the `fetch()`.
